docs/developing/languages/ruby/installation.md

```diff
@@ -16,52 +12,131 @@ for several reasons:
   should be assumed that everything under `/System/Library` is part of macOS and
   is administered by Apple. Files in those directories should not be modified.
 
-- [macOS will no longer include Ruby by default in future releases](https://developer.apple.com/documentation/macos_release_notes/macos_catalina_10_15_beta_9_release_notes).
+## For quick scripts
 
-## Approaches
+It's possible to install Ruby using `homebrew`, however, it's difficult to install precise versions. For one-off uses, though, it's a reasonable approach.
 
-### Limited to one version of Ruby at a time
+## The Trussel Way
 
-- Install Ruby with homebrew
+We recommend installing and managing Ruby using asdf.
 
-- Install Ruby manually
+It's also possible to use Docker, instructions for which are at the bottom of this document.
 
-### Allows switching between multiple versions of Ruby
+### Setup Instructions
 
-- Install and configure a Ruby manager, then install Ruby
+You'll need to know if the project you are running requires a specific version of Ruby. The convention is for this to be defined in a project's `Gemfile`. Look for a line that looks like this:
 
-- Use a vetted automated script that sets everything up for you
+```ruby
+ruby "~> 2.7.2"
+```
 
-## The Trussel Way
+In the above example, any version of Ruby in the range `2.7.2` to `2.7.99999` will suffice.
 
-For long-term day-to-day development, we recommend installing and using Ruby via
-a Ruby manager. Inevitably, when working with Ruby projects, you will need to
-install multiple different versions at the same time and switch between them.
+#### Install Prerequisites
 
-Depending on your preferences and needs, the two recommended ways to install and
-use Ruby on a Mac are via local installation of a Ruby manager, or via Docker.
+- Install [Homebrew](https://brew.sh/)
+- Install Apple Command Line Tools using `xcode-select --install`
 
-Reasons to use Docker:
+#### Install `asdf`
 
-- You don't use Ruby regularly, and only need it occasionally, such as to
-  review candidate work samples
-- You don't want to install anything locally
+[asdf](https://asdf-vm.com) is a program that manages multiple language versions. It uses language-specific plugins to define the specifics for each language it supports. We'll be using [the officially supported plugin for Ruby](https://github.com/asdf-vm/asdf-ruby) in this guide.
+
+Install asdf using homebrew:
+
+```console
+$ brew install asdf
+```
+
+Then, configure your shell to load asdf. The following instructions are for ZSH (the default shell in macOS); for other shells or OSes, find the relevant section in [the asdf installation documentation](https://asdf-vm.com/guide/getting-started.html#_3-install-asdf).
+
+```console
+$ echo -e "\n. $(brew --prefix asdf)/libexec/asdf.sh" >> ${ZDOTDIR:-~}/.zshrc
```

Comment:
On my system, I have `$(brew --prefix asdf)/asdf.sh` (note the missing libexec). Maybe they are the same?

Reply:
That snippet came straight from the asdf docs. Maybe the instructions have changed over time?

On my machine at least, those files are identical:

```console
$ cd $(brew --prefix asdf)
$ pwd
/usr/local/opt/asdf
$ cat asdf.sh
. /usr/local/Cellar/asdf/0.10.2/libexec/asdf.sh
$ diff /usr/local/Cellar/asdf/0.10.2/libexec/asdf.sh libexec/asdf.sh
$
```

Reply:
Wild. I must have misread the docs? or... something? In any case, let's confirm to the docs.